### DISCUSSION TITLE
Page not found, if no base url “/“ and click refresh

### DISCUSSION THREAD

**Discussion by kaisiant** [TanStack/router#412] [Q&A]
I have two routes, “/posts” and “posts/:postId”.
I also have a not found route which will redirect to “/posts”.
All works fine in development. But when deployed to prod, I press refresh on the page “https://domain.com/posts”. The page not found error come out. But if I delete the “/posts” on the url and press enter, the page works again.

Anyone knows what causing this issue? Is it related to js files entry point

**Comment by uxduck** [comment]
It depends how you're hosting your files. Cloudflare Pages, netlify etc. will redirect paths to `/`, so `/posts` should still work. See ann example of a platforms documentation https://developers.cloudflare.com/pages/configuration/serving-pages/#single-page-application-spa-rendering:
> Pages' default single-page application behavior matches all incoming paths to the root (/), allowing you to capture URLs like /about or /help and respond to them from within your SPA.

What server are you using to host your static files?
